src/inx/mod.rs

```diff
@@ -0,0 +1,29 @@
+// Copyright 2022 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+mod error;
+use inx::{client::InxClient, Channel};
+use serde::{Deserialize, Serialize};
+
+pub use self::error::InxError;
+
+/// A builder to establish a connection to INX.
+#[must_use]
+#[derive(PartialEq, Eq, Debug, Serialize, Deserialize)]
+pub struct InxConfig {
```

Comment:
That's not a builder anymore, but the actual config? If so you can remove #[must_use] and adapt the doc.